wgpu-core/src/instance.rs

```diff
@@ -932,6 +939,8 @@ pub enum CreateSurfaceError {
     FailedToCreateSurfaceForAnyBackend(HashMap<Backend, hal::InstanceError>),
     #[error("The display handle used to create this Instance does not match the one used to create a surface on it")]
     MismatchingDisplayHandle,
+    #[error("Neither the `Instance` nor `create_surface()` parameters received a `DisplayHandle`")]
```

Comment:
This is an error that people are going to hit and be extremely confused by, so I'm going to expand this error quite a bit:


```suggestion
    #[error(
        "No `DisplayHandle` is available to create this surface with. When creating a surface with `create_surface()` \
        you must specify a display handle in InstanceDescriptor::display. \
        Rarely, if you need to create surfaces from different `DisplayHandle`s (ex. different wayland or x11 connections), \
        you must use create_surface_unsafe()."
    )]
```